lib/tasks/coupons.rake

```diff
@@ -0,0 +1,24 @@
+namespace :coupons do
+  desc "Create coupons for all the users"
+  task create: :environment do
+    User.find_each do |user|
+      unique_code = loop do
+        random_code = Faker::Code.unique.asin
+        break random_code unless Coupon.exists?(code: random_code)
+      end
+      coupon = user.coupons.create(code: unique_code, description: 'Special Off', start_at: DateTime.current, expire_at: DateTime.current + 1.day)
```

Comment:
what is the difference between Date.current + 1 and Date.current + 1.day

Reply:
No difference.
Added `.day` just to make it clear that 1 day is added.
Or, we can use 1.second, 1.minute to add those respectively.